### PR TITLE
Add data source specific icons to data and join steps in the notebook editor

### PR DESCRIPTION
**Pull request by ranquild**
How to verify:
- New -> Question
- Select a table/question/model/metric
- There should be an icon next to the data source name in DataStep

Screenshot 2024-05-21 at 17 18 48

### CODE REVIEW

**Comment by ranquild** [comment]
@mazameli could you please suggest what the gap between the icon and the label should be?

**xs** (current value):
Screenshot 2024-05-21 at 17 18 48

**sm**:
Screenshot 2024-05-21 at 17 18 34

**Comment by mazameli** [comment]
@ranquild `xs` looks right to me

File: frontend/src/metabase/query_builder/components/notebook/steps/SummarizeStep.tsx [hunk 1]
```diff
@@ -13,14 +13,8 @@ function SummarizeStep({
   step,
   ...props
 }: NotebookStepUiComponentProps) {
-  const isMetric = step.question.type() === "metric";
```
Comment (ranquild) [inline frontend/src/metabase/query_builder/components/notebook/steps/SummarizeStep.tsx:16]: Some leftover from the integration branch. This change never worked and needs to be implemented differently.

**Review by mazameli — COMMENTED** [review]
This is great. How hard would it be to add the `table` and `model` icons to tables and models when you select them? And maybe the `table_grid` icon for saved questions?

**Comment by ranquild** [comment]
@mazameli there is no `table_grid` icon now. What icon should be used instead?

**Comment by mazameli** [comment]
@ranquild sorry, I meant `table2` for questions (we call that one `table_grid` in figma).

**Comment by ranquild** [comment]
@mazameli done

Screenshot 2024-05-21 at 18 08 13
Screenshot 2024-05-21 at 18 08 19
Screenshot 2024-05-21 at 18 08 07
Screenshot 2024-05-21 at 18 10 39

**Review by mazameli — APPROVED** [review]
nice

**Review by kamilmielnik — APPROVED** [review]
@mazameli do we want to display the icon for the joined data as well?

image

**Review by kamilmielnik — APPROVED** [review]
Nice :+1:

**Pull request merged**